src/screens/drive/DrivePreviewScreen/DrivePreviewScreen.tsx

```diff
@@ -0,0 +1,248 @@
+            numberOfLines={2}
+            medium
+          >
+            {filename}
+          </AppText>
+          {!isDownloaded && !error ? (
+            <AppText style={tailwind('text-gray-60 text-center mt-1')}>{getProgressMessage()}</AppText>
+          ) : null}
+        </View>
+        {!isDownloaded && !error ? (
+          <View style={tailwind('w-full mt-10')}>
+            <AppProgressBar height={4} animateWidth currentValue={currentProgress} totalValue={1} />
+          </View>
+        ) : null}
+        {error ? (
+          <View style={tailwind('mt-1')}>
+            <View style={tailwind('flex flex-row items-center')}>
+              <WarningCircle weight="fill" size={20} color={tailwind('text-red-').color as string} />
+              <AppText style={tailwind('text-gray-60 text-center text-red- ml-1')}>{error}</AppText>
+            </View>
+            <AppButton
+              style={tailwind('mt-5')}
+              title={strings.buttons.tryAgain}
+              type={'white'}
+              onPress={() => downloadingFile.retry && downloadingFile.retry()}
+            ></AppButton>
+          </View>
+        ) : null}
+        {isDownloaded && downloadedFilePath ? (
+          <View style={tailwind('mt-1')}>
+            <AppText style={tailwind('text-gray-60 text-center')}>
+              {strings.screens.PreviewScreen.noPreviewAvailable}
+            </AppText>
+            <AppButton
+              style={tailwind('mt-5')}
+              title={strings.screens.PreviewScreen.openWith}
+              type={'white'}
+              onPress={() => handleOpenFileWith(filename, downloadedFilePath)}
+            ></AppButton>
+          </View>
+        ) : null}
+      </View>
+    );
+  };
+  const renderByStatus = () => {
+    // Render no preview
+    if (
+      (currentProgress < 1 && !downloadingFile.downloadedFilePath) ||
+      (downloadingFile.downloadedFilePath && !hasImagePreview && !hasVideoPreview && !hasPdfPreview) ||
+      downloadingFile.error
+    ) {
+      return renderNoPreview({
+        isDownloaded: downloadingFile.downloadedFilePath ? true : false,
+        downloadedFilePath: downloadingFile.downloadedFilePath,
+        error: downloadingFile.error,
+      });
+    }
+
+    // Render an image preview
+    if (downloadingFile.downloadedFilePath && hasImagePreview) {
+      return (
+        <DriveImagePreview
+          onTapImage={() => {
+            setTopbarVisible(!topbarVisible);
+          }}
+          onZoomImage={() => {
+            setTopbarVisible(false);
+          }}
+          imagePath={downloadingFile.downloadedFilePath}
+          height={dimensions.height}
+        />
+      );
+    }
+
+    // Render a video preview
+    if (downloadingFile.downloadedFilePath && hasVideoPreview && generatedThumbnail) {
+      return <DriveVideoPreview thumbnail={generatedThumbnail?.path} source={downloadingFile.downloadedFilePath} />;
+    }
+
+    // Render a PDF preview
```

Comment:
better use a boolean called: 
`const renderPdfPreview = downloadingFile.downloadedFilePath && hasPdfPreview`
in order to avoid comments as code should be as self-explanatory as possible